src/cov.jl

```diff
@@ -44,80 +37,73 @@ that the data are centered and hence there's no need to subtract the mean.
 When `vardim = 1`, the variables are considered columns with observations in rows;
 when `vardim = 2`, variables are in rows with observations in columns.
 """
-function scattermat end
+scattermatm(x::DenseMatrix, mean, vardim::Int=1) =
+    scattermat_zm(x .- mean, vardim)
 
+scattermatm(x::DenseMatrix, mean, wv::AbstractWeights, vardim::Int=1) =
+    scattermat_zm(x .- mean, wv, vardim)
 
-"""
-    cov(X, wv::WeightVec; mean=nothing, vardim=1)
+scattermat(x::DenseMatrix, vardim::Int=1) =
+    scattermatm(x, Base.mean(x, vardim), vardim)
 
-Compute the weighted covariance matrix. By default, the covariance
-matrix is normalized by the sum of the weights. That is, `cov(X, wv)`
-is equivalent to `scattermat(X, wv) / sum(wv)`.
-"""
-cov
+scattermat(x::DenseMatrix, wv::AbstractWeights, vardim::Int=1) =
+    scattermatm(x, Base.mean(x, wv, vardim), wv, vardim)
 
+scattermat_zm(x::DenseMatrix, vardim::Int) = Base.unscaled_covzm(x, vardim)
 
-"""
-    mean_and_cov(x, [wv::WeightVec]; vardim=1) -> (mean, cov)
+scattermat_zm(x::DenseMatrix, wv::AbstractWeights, vardim::Int) =
+    _symmetrize!(Base.unscaled_covzm(x, _scalevars(x, values(wv), vardim), vardim))
 
-Return the mean and covariance matrix as a tuple. A weighting
-vector `wv` can be specified. `vardim` that designates whether
-the variables are columns in the matrix (`1`) or rows (`2`).
 """
-function mean_and_cov end
-
+    cov(X, wv::AbstractWeights, [vardim, corrected])
```

Comment:
The problem with making `corrected` a keyword argument is that I can't deprecate the behaviour because I don't know if corrected has been set. Also, in base `cov` doesn't have `corrected` as a keyword argument, just `std` and `var`.